codalab/rest/bundles.py

```diff
@@ -541,18 +541,22 @@ def _fetch_bundle_contents_blob(uuid, path=''):
     truncation_text = query_get_type(str, 'truncation_text', default='')
     max_line_length = query_get_type(int, 'max_line_length', default=128)
     check_bundles_have_read_permission(local.model, request.user, [uuid])
-    bundle = local.model.get_bundle(uuid)
 
     try:
         target_info = local.download_manager.get_target_info(uuid, path, 0)
+        if target_info['resolved_uuid'] != uuid:
+            uuid = target_info['resolved_uuid']
+            path = target_info['resolved_path']
+            check_bundles_have_read_permission(local.model, request.user, [uuid])
     except NotFoundError as e:
         abort(http.client.NOT_FOUND, str(e))
     except Exception as e:
         abort(http.client.BAD_REQUEST, str(e))
 
     # Figure out the file name.
-    if not path and bundle.metadata.name:
-        filename = bundle.metadata.name
+    bundle_name = local.model.get_bundle(uuid).metadata.name
```

Comment:
Why is a new DB call needed here?  Any performance problems?

Reply:
This is just a moved db call, not an additional one. See the removed db call on line 544 of the before-changes side of the diff. I just realized there was a call to get `bundle` here and it seemed like it was used in multiple places but turns out it was only used here. Kind of a drive-by change.